src/connector/src/source/iceberg/mod.rs

```diff
@@ -110,11 +112,38 @@ impl UnknownFields for IcebergProperties {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct IcebergFileScanTaskJsonStr(String);
+
+impl IcebergFileScanTaskJsonStr {
+    pub fn deserialize(&self) -> FileScanTask {
+        serde_json::from_str(&self.0).unwrap()
+    }
+
+    pub fn serialize(task: &FileScanTask) -> Self {
+        Self(serde_json::to_string(task).unwrap())
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct TableMetadataJsonStr(String);
+
+impl TableMetadataJsonStr {
+    pub fn deserialize(&self) -> TableMetadata {
+        serde_json::from_str(&self.0).unwrap()
+    }
+
+    pub fn serialize(metadata: &TableMetadata) -> Self {
+        Self(serde_json::to_string(metadata).unwrap())
+    }
+}
+
 #[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
 pub struct IcebergSplit {
     pub split_id: i64,
     pub snapshot_id: i64,
-    pub files: Vec<String>,
+    pub table_meta: TableMetadataJsonStr,
+    pub files: Vec<IcebergFileScanTaskJsonStr>,
```

Comment:
Why do we want to serialize them as JSON str, instead of serialize directly?

BTW, for serializing as JSON str, a better way is to `#[serde_as(as = "JsonString")]` https://docs.rs/serde_with/latest/serde_with/json/struct.JsonString.html  
(So we still have deserialized representation in memory)

Reply:
No special purpose. What's the best way to achieve the same goal? Here we just want to serialize the files from frontend and deserialize them from compute.

Reply:
Why not just put the struct (`FileScanTask`) here directly? Then just ser/de the field together when ser/de `IcebergSplit`

Reply:
You are right. Let's refactor it.

Reply:
Oh, I see. Because `Eq, PartialEq` is not implemented for `FileScanTask`

Reply:
`PartialEq` is added to upstream recently. I think we can add these traits if needed.

https://github.com/apache/iceberg-rust/pull/660

Reply:
Hmmm, it seems  Eq and Hash on `IcebergSplit` is unnecessary. After removing them, it can still compile.